digitalocean/datasource_digitalocean_firewall.go

```diff
@@ -0,0 +1,27 @@
+package digitalocean
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceDigitalOceanFirewall() *schema.Resource {
```

Comment:
why return a pointer here? Wouldn't be simpler just to return an actual value?

Reply:
The place that references it (DataSourcesMap on the provider in `digitalocean/provider.go`) is a `map[string]*schema.Resource` and I followed the pattern all the other functions in the provider have already implemented.

Reply:
Right, it needs to implement https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/schema#Provider